src/Microsoft.DotNet.Interactive.Http.Parsing/Parsing/HttpRootSyntaxNode.cs

```diff
@@ -36,11 +38,13 @@ public void Add(HttpRequestSeparatorNode separatorNode)
         AddInternal(separatorNode);
     }
 
-    public Dictionary<string, DeclaredVariable> GetDeclaredVariables()
+    public (Dictionary<string, DeclaredVariable> declaredVariables, List<Diagnostic>? diagnostics) TryGetDeclaredVariables(HttpBindingDelegate? bind = null)
```

Comment:
It's usually good to avoid tuples as return types on public methods. This might also be worth a little discussion.